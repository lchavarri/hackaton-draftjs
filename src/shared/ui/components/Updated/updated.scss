@import '../../assets/variables';

.updated-at {
  display: flex;
  font-size: rem(10px);

  div:last-child {
    margin-left: rem(5px);

    &::before {
      margin-right: rem(5px);
      content: '|';
    }
  }
}
