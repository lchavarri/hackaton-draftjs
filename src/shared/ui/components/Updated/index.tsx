import React from 'react';
import { format, differenceInMilliseconds } from 'date-fns';
import './updated.scss';

type Props = {
  startDate: string | number | Date;
  endDate: string | number | Date;
};

export default function Updated(props: Props) {
  const { startDate, endDate } = props;

  const diff = differenceInMilliseconds(endDate, startDate);
  const duration = !diff || isNaN(diff) ? 0 : diff / 1000;

  return (
    <div className="updated-at">
      <div>Updated: {format(startDate, 'YYYY/MM/DD HH:mm:ss')}</div>
      <div>Duration: {duration}s</div>
    </div>
  );
}
