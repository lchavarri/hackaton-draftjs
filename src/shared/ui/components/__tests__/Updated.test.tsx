import React from 'react';
import { shallow, mount, ReactWrapper } from 'enzyme';
import Updated from '../Updated';

describe('Updated component', () => {
  let component: ReactWrapper;
  let props: any;

  const defaultProps = {
    startDate: new Date(),
    endDate: new Date()
  };
  const getComponent = () => <Updated {...props} />;
  const wrap = () => shallow(getComponent());
  const wrapDeep = () => mount(getComponent());

  beforeEach(() => {
    props = defaultProps;
    component = wrap();
  });

  it('should match component snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
