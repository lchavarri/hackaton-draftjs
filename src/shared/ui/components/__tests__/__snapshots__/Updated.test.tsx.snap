// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Updated component should match component snapshot 1`] = `
<div
  className="updated-at"
>
  <div>
    Updated: 
    2019/11/25 16:18:35
  </div>
  <div>
    Duration: 
    0
    s
  </div>
</div>
`;
